### DISCUSSION TITLE
Describe in docs what is grafana_state_reason: NoData

### DISCUSSION THREAD

**Discussion by lex-pec** [grafana/grafana#68810] [Alerting]
I have some alerts, which by default do not have any data, so their panels show "No Data" in the middle of the graph. 
These alerts are triggered when something appears on the panel - dots of data, usually. So when something appears the alert is triggered and I have a message in my Telegram contact point. When time passes and the time range of the alert leaving that data behind, I am receiving another message, something like: 
``` 
<Alert's name>
grafana_state_reason: NoData

NoData: 
```
Because by default I have no data in my panel a `Nodata = OK` setting is turned on. So I don't understand why I am receiving the latter message.
I've noticed that such a behavior appeared after upgrading Grafana from 9.0.1 to 9.3.6.

Trying to find similar issues or discussions I came across an isssue #57115 which solved a problem of misunderstanding of `grafana_state_reason: MissingSeries`. 

Generally, a similar issue is needed here: to describe how grafana_state_reason: NoData works and the most important is it possible to turn this off and stop receiving this kind of messages when the `NoData = OK` setting is on.

Link to the post on Grafana Community:
https://community.grafana.com/t/alert-returns-grafana-state-reason-nodata-with-nodata-ok-setting/83490

**Comment by lex-pec** [comment]
Sometimes the problem appears even in alerts where the data always persists, as a time series. Additional level of logging and setting Grafana to debug mode didn't give any useful information about the problem.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82578. Feel free to continue the discussion around this there. Thank you!
